plugins/cairo-lang-macro-stable/src/ffi.rs

```diff
@@ -0,0 +1,45 @@
+/// This struct encapsulates a stable ABI representation of struct.
+///
+/// Please be aware that the memory management of values passing the FFI-barrier is tricky.
+/// The memory must be freed on the same side of the barrier, where the allocation was made.
+#[repr(C)]
+#[derive(Debug)]
+#[doc(hidden)]
+pub struct StableSlice<T> {
```

Comment:
If this is taking ownership, then the implementations of Copy and Clone are incorrect because you must clone items, and copy by definition should not exist at all.

If this is taking referefence, then you must protect items with lifetimes